js-src/src/app/services/game.service.js

```diff
@@ -13,8 +13,8 @@ var GameService = Class({
             }
         )
     },
-    getWaitingGames: function () {
-        return this.http.get(this.baseUrl + '?state=waitingForPlayers').map(
+    getWaitingGames: function (diff) {
```

Comment:
i dont think this will change anything, there is no difficulty parameter for get games (and we dont need one do we?)